src/Application/Projects/Commands/Remove/RemoveProjectCommandHandler.cs

```diff
@@ -0,0 +1,26 @@
+using System.Data.Common;
+using MediatR;
+using Projectify.Application.Abstractions.Persistence.Repositories;
+using Projectify.Application.Common.Exceptions;
+
+namespace Projectify.Application.Projects.Commands.Remove;
+
+internal sealed class RemoveProjectCommandHandler(IProjectRepository repository)
+    :IRequestHandler<RemoveProjectCommand, Unit>
+{
+    public async Task<Unit> Handle( 
+        RemoveProjectCommand command,
+        CancellationToken cancellationToken)
+    {
+        var project = await repository.GetByIdAsync(command.Id);
+        if (project == null)
+        {
+            throw new NotFoundException();
+        }
+        var projectId = project.Id;
```

Comment:
[S] Since projectId is used in single place there is no need to assign it to variable, simply pass project.Id into RemoveAsync method